src/styles/popover.scss

```diff
@@ -279,82 +293,151 @@ $block: #{$fd-namespace}-popover;
         border-width: $fd-popover-arrow-height $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half;
         border-top-color: $fd-popover-arrow-color;
 
         &::after {
           bottom: 1px;
-          margin: 0 (-$fd-popover-arrow-width-half);
           border-width: $fd-popover-arrow-height $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half;
           border-top-color: $fd-popover-background-color;
         }
       }
-    }
 
-    &[x-placement^="bottom"] {
-      margin-top: $fd-popover-arrow-height;
-
-      .#{$block}__arrow {
+      &--top {
         top: -$fd-popover-arrow-height;
-        margin: 0 $fd-popover-arrow-width-half;
         border-width: 0 $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half;
         border-bottom-color: $fd-popover-arrow-color;
 
         &::after {
           top: 1px;
-          margin: 0 (-$fd-popover-arrow-width-half);
           border-width: 0 $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half;
           border-bottom-color: $fd-popover-background-color;
         }
       }
-    }
 
-    &[x-placement^="left"] {
-      margin-right: $fd-popover-arrow-height;
-
-      .#{$block}__arrow {
+      &--end {
         right: -$fd-popover-arrow-height;
-        margin: $fd-popover-arrow-width-half 0;
         border-width: $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half $fd-popover-arrow-height;
         border-left-color: $fd-popover-arrow-color;
 
         &::after {
           right: 1px;
-          margin: -$fd-popover-arrow-width-half 0;
           border-width: $fd-popover-arrow-width-half 0 $fd-popover-arrow-width-half $fd-popover-arrow-height;
           border-left-color: $fd-popover-background-color;
         }
       }
-    }
-
-    &[x-placement^="right"] {
-      margin-left: $fd-popover-arrow-height;
 
-      .#{$block}__arrow {
+      &--start {
         left: -$fd-popover-arrow-height;
-        margin: $fd-popover-arrow-width-half 0;
         border-width: $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half 0;
         border-right-color: $fd-popover-arrow-color;
 
         &::after {
           left: 1px;
-          margin: -$fd-popover-arrow-width-half 0;
           border-width: $fd-popover-arrow-width-half $fd-popover-arrow-height $fd-popover-arrow-width-half 0;
           border-right-color: $fd-popover-background-color;
         }
       }
+
+      &-x--start {
+        left: $fd-popover-arrow-offset;
+      }
+
+      &-x--center {
+        left: 50%;
+        transform: translateX(-50%);
+      }
+
+      &-x--end {
+        right: $fd-popover-arrow-offset;
+      }
+
+      &-y--top {
+        top: $fd-popover-arrow-offset;
+      }
+
+      &-y--center {
+        top: 50%;
+        transform: translateY(-50%);
+      }
+
+      &-y--bottom {
+        bottom: $fd-popover-arrow-offset;
+      }
+    }
+
+    .fd-list {
```

Comment:
Please check my comment about the self-contained elements.

Reply:
Agree. But i'm not sure about changing it.

```
// Case 1
<fd-popover>
   <fd-list class="fd-list__popover"> // class can be easily added, refactoring is easy, however it's still the breaking change
   
// Case 2
<ng-template #template>
  // this template is for the popover, but there no information about it, it's handled explicitly
  // it's very hard to refactor
  <fd-list> 
```